New resource: azurerm_express_route_connection (#11320)
terraform-providers/terraform-provider-azurerm: website/docs/r/express_route_connection.html.markdown

---
subcategory: "Network"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_express_route_connection"
description: |-
  Manages an Express Route Connection.
---

# azurerm_express_route_connection

Manages an Express Route Connection.

> ~> **NOTE:** The provider status of the Express Route Circuit must be set as provisioned while creating the Express Route Connection. See more details from https://docs.microsoft.com/en-us/azure/expressroute/expressroute-howto-circuit-portal-resource-manager#send-the-service-key-to-your-connectivity-provider-for-provisioning.

Reviewer: ```suggestion
~> **NOTE:** The provider status of the Express Route Circuit must be set as provisioned while creating the Express Route Connection. See more details [here](https://docs.microsoft.com/en-us/azure/expressroute/expressroute-howto-circuit-portal-resource-manager#send-the-service-key-to-your-connectivity-provider-for-provisioning).
```

Author: Updated

  service_provider_name = "Equinix"
  peering_location      = "Silicon Valley"
  bandwidth_in_mbps     = 50

  sku {
    tier   = "Standard"
    family = "MeteredData"
  }

  allow_classic_operations = false
}

resource "azurerm_express_route_circuit_peering" "example" {
  peering_type                  = "AzurePrivatePeering"
  express_route_circuit_name    = azurerm_express_route_circuit.example.name
  resource_group_name           = azurerm_resource_group.example.name
  shared_key                    = "ItsASecret"
  peer_asn                      = 100
  primary_peer_address_prefix   = "192.168.1.0/30"
  secondary_peer_address_prefix = "192.168.2.0/30"
  vlan_id                       = 100
}

resource "azurerm_express_route_connection" "example" {
  name                             = "example-expressrouteconn"
  express_route_gateway_id         = azurerm_express_route_gateway.example.id
  express_route_circuit_peering_id = azurerm_express_route_circuit_peering.example.id
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Express Route Connection. Changing this forces a new resource to be created.

> * `express_route_circuit_peering_id` - (Required) The ID of the Express Route Circuit Peering within this Express Route Connection should be created. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `express_route_circuit_peering_id` - (Required) The ID of the Express Route Circuit Peering within which this Express Route Connection should be created. Changing this forces a new resource to be created.
```

Author: Updated

> * `express_route_circuit_peering_id` - (Required) The ID of the Express Route Circuit Peering within which this Express Route Connection should be created. Changing this forces a new resource to be created.

Author: Updated

> * `express_route_gateway_id` - (Required) The ID of the Express Route Gateway within this Express Route Connection should be created. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `express_route_gateway_id` - (Required) The ID of the Express Route Gateway within which this Express Route Connection should be created. Changing this forces a new resource to be created.
```

Author: Updated

> * `express_route_gateway_id` - (Required) The ID of the Express Route Gateway within which this Express Route Connection should be created. Changing this forces a new resource to be created.

Author: Updated

* `authorization_key` - (Optional) The authorization key to establish the Express Route Connection.

> * `enable_internet_security` - (Optional) Is internet security enabled for this Express Route Connection?

Author: Updated

* `routing` - (Optional)  A `routing` block as defined below.

> * `routing_weight` - (Optional) The routing weight associated to the Express Route Connection. Valid values are between `0` and `32000`.

Reviewer: We could only assign one value for this, right?
```suggestion
* `routing_weight` - (Optional) The routing weight associated to the Express Route Connection. Valid value is between `0` and `32000`.
```

Author: Updated

> * `routing_weight` - (Optional) The routing weight associated to the Express Route Connection. Valid value is between `0` and `32000`. Defaults to `0`.

Author: Updated
